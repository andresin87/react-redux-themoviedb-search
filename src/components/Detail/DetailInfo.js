import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { List } from 'antd';

import DetailInfoItem from './DetailInfoItem';
import DetailInfoHeader from './DetailInfoHeader';
import { selectorDetail } from '../../reducers/detail';

class DetailInfo extends PureComponent {
  getDataSource = () => {
    const {
      detail: {
        movie: {
          budget,
          genres,
          overview,
          popularity,
          production_companies,
          production_countries,
          release_date,
          runtime,
          spoken_languages,
          status,
          vote_average,
          vote_count,
        },
      },
    } = this.props;

    const dataSource = [
      {
        key: 'genres',
        label: 'Genres',
        type: 'array',
        value: genres,
      },
      {
        key: 'overview',
        label: 'Overview',
        type: 'text',
        value: overview,
      },
      {
        key: 'status',
        label: 'Status',
        type: 'text',
        value: status,
      },
      {
        key: 'release_date',
        label: 'Release date',
        type: 'date',
        value: release_date,
      },
      {
        key: 'runtime',
        label: 'Runtime',
        type: 'time',
        value: runtime,
      },
      {
        key: 'spoken_languages',
        label: 'Spoken languages',
        type: 'array',
        value: spoken_languages,
      },
      {
        key: 'budget',
        label: 'Budget',
        type: 'currency',
        value: budget,
      },
      {
        key: 'production_companies',
        label: 'Prod. companies',
        type: 'array',
        value: production_companies,
      },
      {
        key: 'production_countries',
        label: 'Prod. countries',
        type: 'array',
        value: production_countries,
      },
      {
        key: 'popularity',
        label: 'Popularity',
        type: 'number',
        value: popularity,
      },
      {
        key: 'vote_average',
        label: 'Vote average',
        type: 'number',
        value: vote_average,
      },
      {
        key: 'vote_count',
        label: 'Vote count',
        type: 'number',
        value: vote_count,
      },
    ];

    return dataSource;
  };

  render() {
    const {
      detail: {
        movie: { tagline, title, vote_average },
      },
    } = this.props;
    const dataSource = this.getDataSource();

    return (
      <List
        header={<DetailInfoHeader title={title} tagline={tagline} />}
        size="medium"
        dataSource={dataSource}
        renderItem={item => <DetailInfoItem>{item}</DetailInfoItem>}
      />
    );
  }
}

DetailInfo.propTypes = {
  detail: PropTypes.object.isRequired,
};

const mapStateToProps = state => {
  return { detail: selectorDetail(state) };
};

export default connect(mapStateToProps)(DetailInfo);
