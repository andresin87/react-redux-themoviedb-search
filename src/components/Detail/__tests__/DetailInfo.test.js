import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import DetailInfo from '../DetailInfo';
import store from '../../../config/mockStore';

describe('DetailInfo component', () => {
  const component = (
    <Provider store={store}>
      <DetailInfo />
    </Provider>
  );

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
