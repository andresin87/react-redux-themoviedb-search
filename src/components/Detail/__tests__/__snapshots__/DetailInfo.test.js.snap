// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailInfo component Check the renderItem function 1`] = `
<DetailInfoItem>
  Object {
    "key": "budget",
    "label": "Budget",
    "type": "currency",
    "value": 63000000,
  }
</DetailInfoItem>
`;

exports[`DetailInfo component Snapshot matchs 1`] = `
Array [
  <a
    aria-current="page"
    className="DetailInfoHomeButton active"
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "white",
      }
    }
  >
    Home
  </a>,
  <code
    name="List"
    props={
      Object {
        "dataSource": Array [
          Object {
            "key": "genres",
            "label": "Genres",
            "type": "array",
            "value": Array [
              Object {
                "id": 18,
                "name": "Drama",
              },
            ],
          },
          Object {
            "key": "overview",
            "label": "Overview",
            "type": "text",
            "value": "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground \\"fight clubs\\" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.",
          },
          Object {
            "key": "status",
            "label": "Status",
            "type": "text",
            "value": "Released",
          },
          Object {
            "key": "release_date",
            "label": "Release date",
            "type": "date",
            "value": "1999-10-15",
          },
          Object {
            "key": "runtime",
            "label": "Runtime",
            "type": "time",
            "value": 139,
          },
          Object {
            "key": "spoken_languages",
            "label": "Spoken languages",
            "type": "array",
            "value": Array [
              Object {
                "iso_639_1": "en",
                "name": "English",
              },
            ],
          },
          Object {
            "key": "budget",
            "label": "Budget",
            "type": "currency",
            "value": 63000000,
          },
          Object {
            "key": "production_companies",
            "label": "Prod. companies",
            "type": "array",
            "value": Array [
              Object {
                "id": 508,
                "logo_path": "/7PzJdsLGlR7oW4J0J5Xcd0pHGRg.png",
                "name": "Regency Enterprises",
                "origin_country": "US",
              },
              Object {
                "id": 711,
                "logo_path": "/tEiIH5QesdheJmDAqQwvtN60727.png",
                "name": "Fox 2000 Pictures",
                "origin_country": "US",
              },
              Object {
                "id": 20555,
                "logo_path": null,
                "name": "Taurus Film",
                "origin_country": "",
              },
              Object {
                "id": 54051,
                "logo_path": null,
                "name": "Atman Entertainment",
                "origin_country": "",
              },
              Object {
                "id": 54052,
                "logo_path": null,
                "name": "Knickerbocker Films",
                "origin_country": "",
              },
              Object {
                "id": 25,
                "logo_path": "/qZCc1lty5FzX30aOCVRBLzaVmcp.png",
                "name": "20th Century Fox",
                "origin_country": "US",
              },
              Object {
                "id": 4700,
                "logo_path": "/A32wmjrs9Psf4zw0uaixF0GXfxq.png",
                "name": "The Linson Company",
                "origin_country": "",
              },
            ],
          },
          Object {
            "key": "production_countries",
            "label": "Prod. countries",
            "type": "array",
            "value": Array [
              Object {
                "iso_3166_1": "DE",
                "name": "Germany",
              },
              Object {
                "iso_3166_1": "US",
                "name": "United States of America",
              },
            ],
          },
          Object {
            "key": "popularity",
            "label": "Popularity",
            "type": "number",
            "value": 33.335,
          },
          Object {
            "key": "vote_average",
            "label": "Vote average",
            "type": "number",
            "value": 8.4,
          },
          Object {
            "key": "vote_count",
            "label": "Vote count",
            "type": "number",
            "value": 13618,
          },
        ],
        "header": <DetailInfoHeader
          tagline="Mischief. Mayhem. Soap."
          title="Fight Club"
        />,
        "renderItem": [Function],
        "size": "medium",
      }
    }
  />,
]
`;
